node-services/cards-external-diffusion/src/domain/application/cardsDiffusionRateLimiter.ts

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
+
+
+export default class CardsDiffusionRateLimiter {
+
+    private sendRateLimit: number;
+    private limitPeriodInSec: number = 3600;
+
+    private sentMailPerDestination = new Map();
+
+    public setSendRateLimit(limit: number) {
+        this.sendRateLimit = limit;
+        return this;
+    }
+    
+
+    public setLimitPeriodInSec(period: number) {
+        this.limitPeriodInSec = period;
+        return this;
+    }
+
+    public reset() {
+        this.sentMailPerDestination.clear();
+    }
+
+    public isNewSendingAllowed(destination: string) {
+        const sentMails = this.getSentMails(destination);
+
+        if (!this.isLimitReached(sentMails)) {
+            this.registerNewSending(sentMails);
```

Comment:
i would suggest to call the registerNewSending after mail has been sucessfully sent